ocs_ci/ocs/platform_nodes.py

```diff
@@ -69,7 +69,7 @@ def __init__(self):
 
     def get_nodes_platform(self):
         platform = config.ENV_DATA["platform"]
-        if cluster.is_lso_cluster():
+        if cluster.is_lso_cluster() and platform == constants.VSPHERE_PLATFORM:
```

Comment:
@clacroix12 IIUC, it's because we have a class for vsphere LSO - https://github.com/red-hat-storage/ocs-ci/blob/078610b9996eb5f71e67c8a6b2d12fe3e1943d13/ocs_ci/ocs/platform_nodes.py#L1911
And besides BM which is always LSO based and AWS LSO which so far we haven't implemented platform node class for it, we don't have any other platform that we test platform related operations that is both LSO and dynamic provisioned

Reply:
What is mean is "lso" is not a type of platform so I don't think we should be adding "lso" to the config value for the platform. We have `config.DEPLOYMENT["local_storage"]` which is a boolean that we could use to determine if we are on LSO. We could use both of these together to determine if we are on vsphere and configured for LSO instead of manipulating the platform value like this.

Reply:
The function `is_lso_cluster` is defined here https://github.com/red-hat-storage/ocs-ci/blob/master/ocs_ci/ocs/cluster.py#L1473 and check for the relevant config param. I think it will be better to use that function since we use lots of times the LSO deployment. And about the `platform` param - I guess we don't have to use it, and just pass the value of the class `VMWareLSONodes`

Reply:
I think @clacroix12 has a point. This PR is for fixing https://github.com/red-hat-storage/ocs-ci/issues/3750 and I would like to have it fixed. How about we address the needed change separately?